src/generator/AutoRest.Go/Model/MethodGo.cs

```diff
@@ -112,8 +126,37 @@ public string MethodReturnSignature
       }
     }
 
+    public string MethodReturnSignatureLazy
+    {
+      get
+      {
+        var signature = new StringBuilder("(<- chan ");
+        signature.Append((ListElement.ModelType as SequenceTypeGo).GetElement);
+        signature.Append(", <-chan error)");
+        return signature.ToString();
+      }
+    }
+
+    public PropertyGo ListElement
+    {
+      get
+      {
+        var properties = (ReturnType.Body as CompositeTypeGo).Properties.Cast<PropertyGo>().ToList();
+        foreach (PropertyGo p in properties)
+        {
+          if (p.ModelType is SequenceTypeGo)
+          {
+            return p;
+          }
+        }
+        return null;
+      }
+    }
+
     public string NextMethodName => $"{Name}NextResults";
 
+    public string LazyMethodName => $"{Name}Lazy";
```

Comment:
There are already some `ListAll` funcs  in the swaggers (they are for list by subscription instead than by resource group). To not erpeat funcs names, I chose `ListComplete`, but we can discuss which is a better name.